src/components/BContainer/BContainer.vue

```diff
@@ -0,0 +1,64 @@
+<template>
+  <section role="contentinfo" :aria-label="label">
+    <component v-if=!hideLabel :is=headerElement() class="header">{{ label }}</component>
+    <div>
+      <slot></slot>
+    </div>
+  </section>
+</template>
+
+<script>
+export default {
+  name: 'b-container',
+  props: {
+    label: {
+      type: String,
+      required: true
+    },
+    headerLevel: {
+      type: [String, Number]
+    },
+    hideLabel: {
+      type: Boolean
+    }
+  },
+  created: function() {
+    if (typeof this.headerLevel === "undefined") {
+      console.warn ('No headerLevel property set, defaulting to 2')
+    }
+    if (isNaN(this.headerLevel) || this.headerLevel < 1 || this.headerLevel > 6) {
+      console.error('Invalid headerLevel property set. Valid input are numbers 1-6.  Reverting to default level 2.')
```

Comment:
So we are going to have console warnings instead of throwing errors?

Reply:
Since we can't stop it from mounting we decided to just default to a valid value and log a warning/error.

Reply:
I looked into how Vue does it and they use `console.error`

Reply:
Why are we allowing headerLevel to be a String?  Seems to me we always want it's value to be an int between 1 and 6.  You can also write a validator on it so that the check is cleaner.  Also you could set the default to 2 in the prop. ex:

headerLevel: {
  type: Number,
  default: 2,
  validator (value) {
    if (value < 1 || value > 6)
      throw new TypeError('Pass an actual header')
    return true
  }
}

Reply:
If you don't bind it to a javascript variable and hard code a param it will come across as a string.

For example

<b-container level="3"></b-container> vs <b-container :level="levelVariable">

Reply:
You do have to bind it but this will suffice: <b-container :header-level="3"